packages/serverpod_serialization/lib/src/serialization.dart

```diff
@@ -13,21 +13,25 @@ typedef constructor<T> = T Function(
 /// The [SerializableEntity] is the base class for all serializable objects in
 /// Serverpod, except primitives.
 abstract mixin class SerializableEntity {
-  /// Returns a serialized JSON structure of the model, ready to be sent
-  /// through the API. This does not include fields that are marked as
-  /// database only.
-  dynamic toJson();
-
   /// Returns a serialized JSON structure of the model which also includes
   /// fields used by the database.
-  dynamic allToJson() => toJson();
+  dynamic toJson();
 
   @override
   String toString() {
     return SerializationManager.encode(this);
   }
 }
 
+/// The [ConstraintSerialization] defines a toJson method which makes it
+/// possible to limit what fields are serialized
+abstract interface class ConstraintSerialization {
+  /// Returns a serialized JSON structure of the model, ready to be sent
+  /// through the API. This does not include fields that are marked as
+  /// database only.
+  dynamic toConstraintJson();
+}
+
 /// Get the type provided as an generic. Useful for getting a nullable type.
 Type getType<T>() => T;
 
```

Comment:
Chore: Use the `toConstraintJson` for serialization.

There is one major thing missing with this PR that we didn't catch in testing.

We never actually use the `toConstraintJson` method for serialization.

We need to add tests to ensure we catch this in the future.
We also need to ensure that this flow works with custom classes. I don't think just changing how we serialize on line 169 in this file would fix that. We also need to make sure that we can serialize a custom class that is a field of another class.